generate_stubs.py

```diff
@@ -0,0 +1,180 @@
+        if len(qualified_type) > 0:
+            if imports[qualified_type[0][0]] is None \
+               or qualified_type[0][1] not in imports[qualified_type[0][0]]:
+                module, member = qualified_type[0]
+        else:
+            unqualified_type = re.findall(r'[^\.a-zA-Z0-9_]([A-Z][a-zA-Z0-9_]+)[^\.a-zA-Z0-9_]', ' ' + str(cls) + ' ')
+            if len(unqualified_type) > 0 and unqualified_type[0] != "NoneType":
+                cls_str = str(cls).replace('.en_US', '').replace("faker.", ".")
+                if "<class '" in cls_str:
+                    cls_str = cls_str.split("'")[1]
+                if locale is not None:
+                    cls_str = cls_str.replace('.'+locale, '')
+
+                if imports[cls_str] is None or unqualified_type[0] not in imports[cls_str]:
+                    module, member = cls_str, unqualified_type[0]
+    if module in MODULES_TO_FULLY_QUALIFY:
+        member = None
+    return module, member
+
+
+seen_funcs = set()
+seen_vars = set()
+
+
+class UniqueMemberFunctionsAndVariables:
+    def __init__(self, cls: type, funcs: Dict[str, Any], vars: Dict[str, Any]):
+        global seen_funcs, seen_vars
+        self.cls = cls
+        self.funcs = funcs
+        for func_name in seen_funcs:
+            self.funcs.pop(func_name, None)
+        seen_funcs = seen_funcs.union(self.funcs.keys())
+
+        self.vars = vars
+        for var_name in seen_vars.union(seen_funcs):
+            self.vars.pop(var_name, None)
+        seen_vars = seen_vars.union(self.vars.keys())
+        
+
+def get_member_functions_and_variables(cls: object, include_mangled: bool = False) \
+                                       -> UniqueMemberFunctionsAndVariables:
+    members = [(name, value) for (name, value) in inspect.getmembers(cls) 
+               if ((include_mangled and name.startswith("__")) or not name.startswith("_"))]
+    funcs: Dict[str, Any] = {}
+    vars: Dict[str, Any] = {}
+    for (name, value) in members:
+        attr = getattr(cls, name, None)
+        if attr is not None and (inspect.isfunction(attr) or inspect.ismethod(attr)):
+            funcs[name] = value
+        elif inspect.isgetsetdescriptor(attr) or inspect.ismethoddescriptor(attr):
+            # I haven't implemented logic
+            # for generating descriptor signatures yet
+            continue
+        elif not include_mangled or type(value).__name__ not in GENERIC_MANGLE_TYPES_TO_IGNORE:
+            vars[name] = value
+
+    return UniqueMemberFunctionsAndVariables(cls, funcs, vars)
+
+
+classes_and_locales_to_use_for_stub: List[Tuple[object, str]] = []
+for locale in AVAILABLE_LOCALES:
+    for provider in PROVIDERS:
+        if provider == "faker.providers":
+            continue
+        prov_cls, _, _ = Factory._find_provider_class(provider, locale)
+        classes_and_locales_to_use_for_stub.append((prov_cls, locale))
+
+all_members: List[Tuple[UniqueMemberFunctionsAndVariables, str]] = \
```

Comment:
Should be `List[Tuple[UniqueMemberFunctionsAndVariables, Optional[str]]]` since the last item in the list has `None` in it.